coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -270,3 +273,61 @@ def get_messages(self, dataset_id, last_updated_after=None):
                 seen_message_ids.add(message.message_id)
 
             return messages
+
+    def get_code_schemes_ref(self, segment_id):
+        """
+        Gets Firestore database reference to code schemes.
+
+        :param segment_id: Id of a segment.
+        :type segment_id: str
+        :return: A reference to collection `code_schemes` in Firestore database
+        :rtype: google.cloud.firestore_v1.collection.CollectionReference
+        """
+        return self._client.collection(f"datasets/{segment_id}/code_schemes")
+
+    def ensure_code_schemes_consistent(self, dataset_id):
+        """
+        Checks that the code schemes are the same in all segments
+
+        :param dataset_id: Id of a dataset.
+        :type dataset_id: str
+        """
+        segment_count = self.get_segment_count(dataset_id)
+        if segment_count is None or segment_count == 1:
+            return
+
+        first_segment_schemes = []
+        for code_scheme in self.get_code_schemes_ref(dataset_id).get():
+            first_segment_schemes.append(CodeScheme.from_firebase_map(code_scheme.to_dict()))
+
+        for segment_index in range(2, segment_count + 1):
+            segment_id = self.id_for_segment(dataset_id, segment_index)
+
+            current_segment_schemes = []
+            for code_scheme in self.get_code_schemes_ref(segment_id).get():
+                current_segment_schemes.append(CodeScheme.from_firebase_map(code_scheme.to_dict()))
+
+            assert len(first_segment_schemes) == len(current_segment_schemes), \
+                f"Segment {segment_id} has a different number of schemes to the first segment {dataset_id}"
+
+            first_segment_schemes.sort(key=lambda s: s.scheme_id)
+            current_segment_schemes.sort(key=lambda s: s.scheme_id)
+
+            for x, y in zip(first_segment_schemes, current_segment_schemes):
+                assert x == y, f"Segment {segment_id} has different schemes to the first segment {dataset_id}"
+
+    def get_all_code_schemes(self, dataset_id):
+        """
+        Gets all code schemes for a given dataset
+
+        :param dataset_id: Id of a dataset.
+        :type dataset_id: str
+        :return: Code schemes in this dataset
+        :rtype: core_data_modules.data_models.code_scheme.CodeScheme
+        """
+        self.ensure_code_schemes_consistent(dataset_id)
+
+        code_schemes = []
+        for code_scheme in self.get_code_schemes_ref(dataset_id).get():
```

Comment:
I would call this doc as calling it a code_scheme applies it has type CodeScheme, which it doesn't

Reply:
Thanks, updated.